Arrowgene.Ddon.GameServer/Handler/InstanceEnemyKillHandler.cs

```diff
@@ -137,38 +138,28 @@ public override S2CInstanceEnemyKillRes Handle(GameClient client, C2SInstanceEne
                     queuedPackets.Send();
                 }
 
-                var dropItemNtc = new S2CInstancePopDropItemNtc()
-                {
-                    LayoutId = packet.LayoutId,
-                    SetId = packet.SetId,
-                    MdlType = enemyKilled.DropsTable?.MdlType ?? 0,
-                    PosX = packet.DropPosX,
-                    PosY = packet.DropPosY,
-                    PosZ = packet.DropPosZ
-                };
                 foreach (var partyMemberClient in client.Party.Clients)
                 {
-                    // If the enemy is quest controlled, then either get from the quest loot drop, or the general one.
-                    List<InstancedGatheringItem> instancedGatheringItems = new List<InstancedGatheringItem>();
-
-                    // Items from kill an enemy normally
-                    instancedGatheringItems.AddRange(IsQuestControlled ?
-                                partyMemberClient.InstanceQuestDropManager.GenerateEnemyLoot(quest, enemyKilled, packet.LayoutId, packet.SetId) :
-                                partyMemberClient.InstanceDropItemManager.GetAssets(layoutId, (int)packet.SetId));
+                    var instancedGatheringItems = partyMemberClient.InstanceDropItemManager.Generate(enemyKilled);
 
-                    // Items for any server events which might be active
-                    instancedGatheringItems.AddRange(partyMemberClient.InstanceEventDropItemManager.GenerateEventItems(partyMemberClient, enemyKilled, packet.LayoutId, packet.SetId));
+                    uint offsetSetId = partyMemberClient.InstanceDropItemManager.Assign(layoutId, packet.SetId, instancedGatheringItems.Values.SelectMany(x => x).ToList());
+                    var dropItemNtc = new S2CInstancePopDropItemNtc()
+                    {
+                        LayoutId = packet.LayoutId,
+                        SetId = offsetSetId,
```

Comment:
If i understand correctly, the drop SetId value is entirely unrelated to the enemy SetId value (which would be the position index), right? So if a player were to kill enemies in positions in order 3 1 2 we could send drop set ids 1 2 3 and it'd work out fine, removing the need to handle the offsetting if there's a collision.
I'm not sure if this would make the code cleaner or if it would complicate things by having SetId mean two different things, but it's worth considering.